pkg/tests/api/alerting/api_provisioning_test.go

```diff
@@ -0,0 +1,157 @@
+package alerting
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/tests/testinfra"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProvisioning(t *testing.T) {
+	_, err := tracing.InitializeTracerForTest()
+	require.NoError(t, err)
+
+	dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+		DisableLegacyAlerting: true,
+		EnableUnifiedAlerting: true,
+		DisableAnonymous:      true,
+		AppModeProduction:     true,
+		EnableFeatureToggles:  []string{featuremgmt.FlagAlertProvisioning},
+	})
+
+	grafanaListedAddr, store := testinfra.StartGrafana(t, dir, path)
+	// override bus to get the GetSignedInUserQuery handler
+	store.Bus = bus.GetBus()
+
+	// Create a users to make authenticated requests
+	createUser(t, store, models.CreateUserCommand{
+		DefaultOrgRole: string(models.ROLE_VIEWER),
+		Password:       "viewer",
+		Login:          "viewer",
+	})
+	createUser(t, store, models.CreateUserCommand{
+		DefaultOrgRole: string(models.ROLE_EDITOR),
+		Password:       "editor",
+		Login:          "editor",
+	})
+	createUser(t, store, models.CreateUserCommand{
+		DefaultOrgRole: string(models.ROLE_ADMIN),
+		Password:       "admin",
+		Login:          "admin",
+	})
+
+	t.Run("when provisioning notification policies", func(t *testing.T) {
```

Comment:
@alexweav Can we add a test with an invalid payload?

Reply:
Another test I would like to see is that if something got provisioned from source `a`, it can not be changed by source `b`. Perhaps this is something you already got in mind for the Admin-API so I'm not sure.

Reply:
Discussed offline, will come in the next PR.

Reply:
I will address these two in future PRs:
1. The next PR to extract validators will cover this. Right now, we do no validation (I left a TODO)
2. Discussed offline - we aren't handling that header yet. this will be another PR to set this behavior across all routes, as this also depends on the existing POST AM config route. Hardcoding API Provenance is a sane stop-gap default for the provisioning API.